e2e/e2e-create-windows-nodepool.sh

```diff
@@ -6,13 +6,24 @@ source e2e-helper.sh
 
 log "Starting to create windows nodepool"
 
-out=$(az aks nodepool list --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME | jq '.[1].name')
+if [[ "$RESOURCE_GROUP_NAME" == *"windows"*  ]]; then
+    RESOURCE_GROUP_NAME="$RESOURCE_GROUP_NAME"-"$WINDOWS_E2E_IMAGE"
+fi
+
+az aks nodepool show --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME --name "winnp"
+retval=$?
 
-if [ "$out" == null ]; then
+if [[ "$retval" != "0" ]]; then
     log "Creating windows nodepool"
-    # If you want to specify the k8s version for windows nodepool, you should also specify the version when creating cluster
-    # and modify the value of "customKubeBinaryURL" and "customKubeProxyImage" in property-windows.json
-    az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name $WINDOWS_NODEPOOL --os-type Windows --node-count 1
+    if [ "$WINDOWS_E2E_IMAGE" == "2019-containerd" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --node-count 1
+    elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd" ]; then
+        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-count 1
```

Comment:
Does the default VMSize only support gen1?

Reply:
No, default VMSize is Standard_D2s_v3. It supports both gen1 and gen2. Should I use a VMSize that only supports gen1 as default VMSize?

Reply:
yes. otherwise ws2022 gen1 will not be covered.